tsdb/querier.go

```diff
@@ -106,14 +109,29 @@ func (q *querier) lvals(qs []Querier, n string) ([]string, error) {
 }
 
 func (q *querier) Select(ms ...*labels.Matcher) (SeriesSet, error) {
+	if len(q.blocks) != 1 {
+		return q.SelectSorted(ms...)
+	}
+	// Sorting Head series is slow, and unneeded when only the
+	// Head is being queried. Sorting blocks is a noop.
+	return q.blocks[0].Select(ms...)
+}
+
+func (q *querier) SelectSorted(ms ...*labels.Matcher) (SeriesSet, error) {
 	if len(q.blocks) == 0 {
 		return EmptySeriesSet(), nil
 	}
 	ss := make([]SeriesSet, len(q.blocks))
 	var s SeriesSet
 	var err error
 	for i, b := range q.blocks {
-		s, err = b.Select(ms...)
+		if len(q.blocks) == 1 {
```

Comment:
When we call this function and q.blocs is 1 ?

Reply:
Ah, that's from when I refactored this a bit. Let me fix that.

Reply:
if the function is exposed publicly that might still make sense

Reply:
If you call the Sorted version of the function, the result should still be sorted.

Reply:
As we just have one block arent the postings sorted already?